### DISCUSSION TITLE
Local video file - transcript

### DISCUSSION THREAD

**Discussion by Auss13** [danielmiessler/Fabric#1152] [Q&A]
Hi everyone, 
just installed and set Fabric UP! so this is a basic question ...

I am trying to summarize a meeting recording but I can't get it to work.
![image](https://github.com/user-attachments/assets/1415e550-e204-4b69-883a-079baaf64932)

form the shell:
**fabric --transcript  "./20241119_140457-MeetingRecording.mp4" "can you summarize  this video"
Please provide me with the link to the video you would like me to summarize! 😊

I need the URL to access and understand the content.**

What am I doing wrong?
and yes, the video is indeed there

Any help is really appreciated.
Thanks
Matt

**Comment by mseigal** [comment]
I use whisper ai from OpenAI. It is an open source transcription utility that runs on the device and works with ffmpeg to throw media files.  My use case was to automate transcriptions and I use the tiny model with Whisper on a 2021 M1 Mac mini. 

Once whisper is installed I transcribe a media file with a command such as ~/whisper/recording.wav --output_format txt --model tiny --output recording.txt && cat recording.txt | fabric -p summarize

recording.wav can be substituted with your recording media file. Whisper will generate a file called recording.txt but that can be changed and the next part will send the contents of this file to fabric for further processing using whatever pattern you wish. In this case I am using summarize. 

I guess there are more elegant solutions but this worked for me.

**Reply by yousecjoe** [reply]
I was not aware that Fabric worked on local videos. 

For Fabric to use a video, it must be on YouTube unless I'm mistaken. 

The URL you need is the one from YT.

**Reply by mseigal** [reply]
It is technically true that you can only generate transcriptions using the -y flag by applying it to YouTube links. Local videos first need to be transcribed using other tools like whisper ai, which can be piped through to fabric for subsequent ai processing.
